R/parents.R

```diff
@@ -0,0 +1,147 @@
+#' Getter for specific parent
+#'
+#' @param join_keys_obj (`JoinKeys`) object to retrieve.
+#' @param dataset_name (`character(1)`)
+#'
+#' @export
+#'
+#' @examples
+#' jk <- join_keys()
+#' parent(jk, "ds1")
+#' parents(jk) <- list("ds2" = "ds3")
+#' parent(jk, "ds2")
+parent <- function(join_keys_obj, dataset_name) {
+  if (missing(dataset_name)) {
+    return(NULL)
+  }
+  checkmate::assert_string(dataset_name)
+  # assert join_keys_obj is performed by parents()
+  parents(join_keys_obj)[[dataset_name]]
+}
+
+#' Getter and setter functions for parents attribute of `JoinKeys`
+#'
+#' @param join_keys_obj (`JoinKeys`) object to retrieve or manipulate.
+#' @return a list of `character` representing the parents.
+#'
+#' @export
+parents <- function(join_keys_obj) {
+  UseMethod("parents", join_keys_obj)
+}
+
+#' @rdname parents
+#' @export
+#' @examples
+#' jk <- join_keys()
+#' parents(jk)
+parents.JoinKeys <- function(join_keys_obj) {
+  rlang::`%||%`(attr(join_keys_obj, "__parents__"), list())
```

Comment:
Please either use `if` or import `%||%`

Reply:
Done in af7dbf6e. Imported `%||%` as it's used twice.